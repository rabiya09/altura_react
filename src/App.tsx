import React from "react";
import { BrowserRouter, BrowserRouter as Router } from "react-router-dom";
import Footer from "./pages/common/Footer";
import Header from "./pages/common/Header";
import Nav from "./pages/common/Nav";
import Routers from "./router";

function App() {
  return (
    <>
    
    <div className="App">
    <BrowserRouter>
      <Header/>
    <Nav/>
        <Routers />
        <Footer/>
        </BrowserRouter>
      
    </div>
   
    </>
  );
}

export default App;
