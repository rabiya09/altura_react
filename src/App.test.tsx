import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App';

/* test.only('renders app name', () => {
  render(<App />);
  const linkElement = screen.getByTestId('app-name');
  expect(linkElement).toBeInTheDocument();
}); */
describe('rabiya', () => {
  test('should match snapshot', () => {
    const {asFragment} = render(<App />);
    expect(screen.getAllByText(/Register/i)[0]).toBeInTheDocument();
   // expect(screen.getByTestId('app-name')).toBeInTheDocument();
    
    expect(asFragment()).toMatchSnapshot();
  })

});