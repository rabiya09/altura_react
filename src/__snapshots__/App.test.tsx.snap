// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rabiya should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="header-bar-container"
    >
      <div
        class="header-bar"
      >
        <div
          class="header-logo-container"
        >
          <span
            class="header-logo-image"
          >
            <a
              title="Altura"
            >
              <img
                alt="Altura"
                src="logo.svg"
              />
            </a>
          </span>
          <span
            class="header-logo-text"
          >
            <a
              data-testid="app-name"
              title="Altura"
            >
              ALTURA
            </a>
          </span>
        </div>
        <div
          class="header-menu-container"
        >
          <div
            class="dropdown mobiledropdown"
          >
            <button
              class="dropbtn mobile-view"
            >
              <div
                class="hamburger-menu"
              >
                <div
                  class="hamburger"
                />
                <div
                  class="hamburger"
                />
                <div
                  class="hamburger"
                />
              </div>
            </button>
            <div
              class="dropdown-content"
            >
              <a>
                Register
              </a>
              <a>
                Products
              </a>
              <a>
                Dashboard
              </a>
              <a>
                Cart
              </a>
              <a>
                Logout
              </a>
            </div>
          </div>
          <ul
            class="header-menu-ul"
          >
            <li
              class="register-li"
            >
              <a
                title="Register"
              >
                Register
              </a>
            </li>
            <li
              class="product-details-li"
            >
              <a
                title="Product Details"
              >
                Product Details
              </a>
            </li>
          </ul>
          <div
            class="search-container"
          >
            <input
              placeholder="search"
              title="Search"
              type="text"
            />
          </div>
          <div
            class="icon-container"
          >
            <ul>
              <li
                class="user-icon"
              >
                <div
                  class="dropdown"
                >
                  <button
                    class="dropbtn"
                  >
                    <img
                      alt="User Details"
                      src="account_circle.svg"
                      title="User Details"
                    />
                  </button>
                  <div
                    class="dropdown-content"
                  >
                    <a>
                      Dashboard
                    </a>
                    <a>
                      Cart
                    </a>
                    <a>
                      Logout
                    </a>
                  </div>
                </div>
              </li>
              <li>
                <a>
                  <img
                    alt="Call"
                    src="call.svg"
                    title="Call"
                  />
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
    <div
      class="nav-bar-container"
    >
      <div
        class="nav-bar"
      >
        <ul>
          <li
            class="new-in-li"
          >
            <span
              class="new-in-li-image"
            >
              <a
                title="New In"
              >
                New In
              </a>
            </span>
          </li>
          <li
            class="chair-li"
          >
            <span
              class="chair-li-image"
            >
              <img
                alt="Chairs"
                src="Chair.svg"
              />
            </span>
          </li>
          <li
            class="sofa-li"
          >
            <span
              class="sofa-li-image"
            >
              <img
                alt="Sofas"
                src="Sofa.svg"
              />
            </span>
          </li>
          <li
            class="table-li"
          >
            <span
              class="table-li-image"
            >
              <img
                alt="Tables"
                src="icon-Tables.svg"
              />
            </span>
            <span
              class="table-li-text"
            >
              Tables
            </span>
          </li>
          <li
            class="lighting-li"
          >
            <span
              class="lighting-li-image"
            >
              <img
                alt="Lighting"
                src="Lighthing.svg"
              />
            </span>
          </li>
        </ul>
      </div>
    </div>
    <div>
      <div
        class="banner-container"
      >
        <div
          class="banner"
        >
          <div
            class="office-chair-rate"
          >
            Office Chair from
            <span>
              $50
            </span>
          </div>
          <div
            class="office-chair-image"
          >
            <img
              alt="Banner - Chair"
              src="banner-image 2.svg"
            />
          </div>
          <div
            class="middle-banner"
          >
            2 great furniture on sale!
            <span>
              Deal will close in 2 days
            </span>
          </div>
          <div
            class="sectionals-rate"
          >
            Sectionals
            <span>
               from
            </span>
            <span>
              $100
            </span>
          </div>
          <div
            class="sectionals-image"
          >
            <img
              alt="Banner - Sofa"
              src="banner-image 1.svg"
            />
          </div>
        </div>
      </div>
      <div
        class="new-in-container"
      >
        <div
          class="new-in"
        >
          New In
        </div>
        <div
          class="product-box-card-container"
        >
          <div
            class="product-box-card"
          >
            <div
              class="productcard-img-box"
            >
              <a
                href="/product-details/1"
              >
                <img
                  alt="Chair1"
                  class="productcard-img"
                  src="../../images/image 13.svg"
                />
              </a>
            </div>
            <div
              class="product-name"
            >
              <a
                href="/product-details/1"
              >
                Chair1
              </a>
            </div>
            <div
              class="product-price"
            >
              170.00
            </div>
          </div>
          <div
            class="product-box-card"
          >
            <div
              class="productcard-img-box"
            >
              <a
                href="/product-details/2"
              >
                <img
                  alt="Bed1"
                  class="productcard-img"
                  src="../../images/image 4.svg"
                />
              </a>
            </div>
            <div
              class="product-name"
            >
              <a
                href="/product-details/2"
              >
                Bed1
              </a>
            </div>
            <div
              class="product-price"
            >
              1000.00
            </div>
          </div>
          <div
            class="product-box-card"
          >
            <div
              class="productcard-img-box"
            >
              <a
                href="/product-details/3"
              >
                <img
                  alt="Chair2"
                  class="productcard-img"
                  src="../../images/image 2.svg"
                />
              </a>
            </div>
            <div
              class="product-name"
            >
              <a
                href="/product-details/3"
              >
                Chair2
              </a>
            </div>
            <div
              class="product-price"
            >
              302.00
            </div>
          </div>
          <div
            class="product-box-card"
          >
            <div
              class="productcard-img-box"
            >
              <a
                href="/product-details/4"
              >
                <img
                  alt="Chair3"
                  class="productcard-img"
                  src="../../images/image 3.svg"
                />
              </a>
            </div>
            <div
              class="product-name"
            >
              <a
                href="/product-details/4"
              >
                Chair3
              </a>
            </div>
            <div
              class="product-price"
            >
              279.00
            </div>
          </div>
        </div>
      </div>
      <div
        class="hr-line"
      >
        <hr />
      </div>
      <div
        class="home-office-container"
      >
        <div
          class="home-office"
        >
          <div
            class="home-office-text"
          >
            <span
              class="home-office-title"
              data-testid="page-title"
            >
              Home Office
            </span>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
            <span
              class="read-more"
            >
              <a
                href="/shop-all-home-office"
              >
                Shop All
              </a>
            </span>
          </div>
          <div
            class="home-office-image"
          >
            <img
              alt="Home Office"
              src="homeoffice.svg"
            />
          </div>
        </div>
      </div>
      <div
        class="bed-room-container"
      >
        <div
          class="bed-room"
        >
          <div
            class="bed-room-image"
          >
            <img
              alt="Bed Room"
              src="bedroom.svg"
            />
          </div>
          <div
            class="bed-room-text"
          >
            <span
              class="bed-room-title"
              data-testid="page-title"
            >
              Bed Room
            </span>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
            <span
              class="read-more"
            >
              <a
                href="/shop-all-bed-room"
              >
                Shop All
              </a>
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="footer-box"
    >
      <div
        class="footer-logo-box"
      >
        <div
          class="footer-logo"
        >
          <img
            alt="Footer Logo"
            src="LogoFooter.svg"
          />
          <span
            class="footer-logo-text"
          >
            ALTURA
          </span>
        </div>
      </div>
      <div
        class="footer-link-first footer-links"
      >
        <ul>
          <li>
            <a
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              href="/"
            >
              Fill Form
            </a>
          </li>
          <li>
            <a
              href="/"
            >
              See Other Details
            </a>
          </li>
          <li>
            Our Products
          </li>
          <li>
            Help
          </li>
        </ul>
      </div>
      <div
        class="footer-link-second"
      >
        <ul>
          <li>
            <a
              href="/"
            >
              New In
            </a>
          </li>
          <li>
            <a
              href="/sofas"
            >
              Sofas
            </a>
          </li>
          <li>
            <a
              href="/tables"
            >
              Tables
            </a>
          </li>
          <li>
            <a
              href="/chairs"
            >
              Chairs
            </a>
          </li>
          <li>
            <a
              href="/lighting"
            >
              Lighting
            </a>
          </li>
        </ul>
      </div>
      <div
        class="footer-link-third"
      >
        <ul>
          <li>
            Visit Our Stores
          </li>
          <li>
            UB City
          </li>
          <li>
            Sun City
          </li>
          <li>
            White Town
          </li>
          <li>
            Cape City
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
